classes/security/authorization/internal/SubmissionAuthorPolicy.inc.php

```diff
@@ -45,13 +46,27 @@ function effect() {
 		$submission = $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION);
 		if (!is_a($submission, 'Submission')) return AUTHORIZATION_DENY;
 
+		$context = $this->_request->getContext();
+
 		// Check authorship of the submission. Any ROLE_ID_AUTHOR assignment will do.
 		$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
 		$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
 		$submitterAssignments = $stageAssignmentDao->getBySubmissionAndStageId($submission->getId(), null, null, $user->getId());
+		$workflowStages = Application::getApplicationStages();
+		$userService = Services::get('user');
```

Comment:
Although I started this when we first introduced services, I am trying to steer clear of assigning it to a variable. There's no extra overhead calling `Services::get()` so whenever using it I prefer just calling it directly.

So below it would be:

```
Services::get('user')->getAccessibleStageRoles(...)
```

I feel that it is easier to track down uses of the service class this way and it adds very little in terms of extra code length.

Reply:
check